tests/Namshi/JOSE/Test/JWSTest.php

```diff
@@ -44,6 +45,20 @@ public function testValidationOfAValidJWS()
         $this->assertTrue($jws->isValid($public_key));
     }
 
+    public function testUseOfCustomEncoder()
+    {
+        $encoder = $this->prophesize('Namshi\JOSE\Base64\Encoder');
+        $encoder
+            ->decode(Argument::any())
+            ->willReturn('{"whatever": "the payload should be"}')
+            ->shouldBeCalled();
+        $encoder
+            ->decode(Argument::any())
+            ->willReturn('{"alg": "test"}')
+            ->shouldBeCalled();
+        JWS::load($this->jws->getTokenString(), $encoder->reveal());
+    }
```

Comment:
No, the same method is mocked for two consecutive calls (decoding the header then decoding the payload). Apparently, this need to be done in reverse order.